sdk/storage/sample/datalake_getting_started.cpp

```diff
@@ -3,11 +3,100 @@
 
 #include "datalake/datalake.hpp"
 #include "samples_common.hpp"
+#include <iostream>
 
 SAMPLE(DataLakeGettingStarted, DataLakeGettingStarted)
 void DataLakeGettingStarted()
 {
+  // Please note that you can always reference test cases for advanced usages.
+
   using namespace Azure::Storage::Files::DataLake;
-  auto client = ServiceClient::CreateFromConnectionString(GetConnectionString());
-  auto response = client.ListFileSystems();
+
+  std::string fileSystemName = "sample-file-system";
+  std::string directoryName = "sample-directory";
+  std::string fileName = "sample-file";
+
+  // Initializing a ServiceClient that can then initialize the FileSystemClient or list file
+  // systems.
+  auto serviceClient = ServiceClient::CreateFromConnectionString(GetConnectionString());
+
+  try
+  {
+    // List all file systems.
+    std::string continuation;
+    std::vector<FileSystem> fileSystems;
+    do
+    {
+      auto response = serviceClient.ListFileSystems();
+      if (response->Continuation.HasValue())
+      {
+        continuation = response->Continuation.GetValue();
+      }
+      fileSystems.insert(
+          fileSystems.end(), response->Filesystems.begin(), response->Filesystems.end());
+    } while (!continuation.empty());
+
+    // Find if the file systems already exist.
+    auto fileSystemClient = serviceClient.GetFileSystemClient(fileSystemName);
+    auto iter = std::find_if(
+        fileSystems.begin(), fileSystems.end(), [&fileSystemName](const FileSystem& fileSystem) {
+          return fileSystem.Name == fileSystemName;
+        });
+    // Create file systems if not exist.
+    if (iter == fileSystems.end())
+    {
+      fileSystemClient.Create();
+    }
+
+    // Create a directory.
+    auto directoryClient = fileSystemClient.GetDirectoryClient(directoryName);
+    directoryClient.Create();
+
+    // Creates a file under the directory.
+    auto fileClient = directoryClient.GetFileClient(fileName);
+    fileClient.Create();
+
+    // Append/flush/read data from the client.
+    // Append data
+    std::string str1 = "Hello ";
+    std::string str2 = "World!";
+    std::vector<uint8_t> buffer(str1.begin(), str1.end());
```

Comment:
You're making a copy of all the data to be uploaded, this can be misleading. Because several of our Track1 customers ever had this concern of too many copies.

Reply:
This is just to show that our customers can also use the buffer if they want to. We shouldn't care how the buffer ifsconstructed and customers shouldn't reference this sample code to see how the raw data, which is customer input, is constructed.

Reply:
Then how did you tell our customers that they shouldn't reference this sample?

Reply:
Well, it's to me a common sense not to reference how *input* parameters are initialized for any samples. If we should also be careful about that, then should we also comment on `GetConnectionString()` not to be the recommended way of setting/getting connection strings?

Reply:
Sorry, but this doesn't look like a common sense to me.

If you have any good suggestions about how to set connection string in sample, we can discuss about it.

Reply:
I'm not saying current `GetConnectionString()`, what track 1 does (hard-coded), or what we are doing here (getting from system env) are both fine to me as it is an input parameter. I don't think we should go an extra mile to regulate what customer tries to do with input parameter which is transparent to storage SDK as long as the type is right and data is valid. 
> Because several of our Track1 customers ever had this concern of too many copies.

Track 1 SDK customer complains about the copies within the SDK, I don't think it's rational to complain about copies in customer code, which is managed by themselves.
>Sorry, but this doesn't look like a common sense to me.

Well, if you don't think it's common sense, I can somehow understand why I should comment that this is not what we suggest our customer to do. Added the comment for clarity.